### PR TITLE
Render groups badges at top of source page

### PR DESCRIPTION
**Pull request by acrellin**
image

### CODE REVIEW

**Comment by acrellin** [comment]
[ch1098]

File: skyportal/handlers/api/source.py [hunk 1]
```diff
@@ -192,6 +192,13 @@ def get(self, obj_id=None):
             s.classifications = s.get_classifications_owned_by(self.current_user)
             source_info = s.to_dict()
             source_info["last_detected"] = s.last_detected
+            source_info["groups"] = Group.query.filter(
+                Group.id.in_(
+                    DBSession().query(Source.group_id).filter(
+                        Source.obj_id == obj_id
+                    )
+                )
+            ).all()
```
Comment (dannygoldstein) [inline skyportal/handlers/api/source.py:201]:
```suggestion
            ).filter(Group.id.in_(self.current_user.accessible_groups)).all()
```

Comment (acrellin) [reply skyportal/handlers/api/source.py:201]: Really? People can't even see which other groups it belongs to if they're not a member of those??

Comment (acrellin) [reply skyportal/handlers/api/source.py:203]: @dannygoldstein Done

**Comment by profjsb** [comment]
Also, we dont have a restriction on group name length. So you might want to cut off the badge size to <= 10 characters (ie. "'Fabulous Green Transie...")

**Comment by acrellin** [comment]
> Looks good. The badge being there or not there (following group accessibility) should be added to the tests. Also, have you looked at the UI when the source belongs to many groups (like dozens?) ... will they wrap to another line? Will you show in a carousel?

@profjsb I've updated the tests to check that the badges render. In the case of many groups, they'll just wrap like this: 
image

Do you think there's a need to contain it in a scrollable div with max height? Not sure how many groups a typical source will be associated with...

**Comment by acrellin** [comment]
@profjsb I've also truncated long group names in the badges

**Pull request merged**